src/LCRmeter-Metrohm_AutolabPGSTAT/main.py

```diff
@@ -0,0 +1,333 @@
+# This Device Class is published under the terms of the MIT License.
+# Required Third Party Libraries, which are included in the Device Class
+# package for convenience purposes, may have a different license. You can
+# find those in the corresponding folders or contact the maintainer.
+#
+# MIT License
+#
+# Copyright (c) 2024 SweepMe! GmbH (sweep-me.net)
+#
+# Permission is hereby granted, free of charge, to any person obtaining a copy
+# of this software and associated documentation files (the "Software"), to deal
+# in the Software without restriction, including without limitation the rights
+# to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+# copies of the Software, and to permit persons to whom the Software is
+# furnished to do so, subject to the following conditions:
+#
+# The above copyright notice and this permission notice shall be included in all
+# copies or substantial portions of the Software.
+#
+# THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+# IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+# FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+# AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+# LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+# OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+# SOFTWARE.
+
+
+# SweepMe! driver
+# * Module: LCRmeter
+# * Instrument: Metrohm Autolab PGSTAT
+
+from __future__ import annotations
+
+import time
+from pathlib import Path
+
+import clr
+from pysweepme import debug
+from pysweepme.EmptyDeviceClass import EmptyDevice
+
+autolab_sdk_path = r"C:\Program Files\Metrohm Autolab\Autolab SDK 2.1\EcoChemie.Autolab.Sdk.dll"
+extension_path = r"C:\Program Files\Metrohm Autolab\Autolab SDK 2.1\EcoChemie.Autolab.SDK.Extensions.dll"
+
+import_failed = False
+try:
+    clr.AddReference(autolab_sdk_path)
+    clr.AddReference(extension_path)
+    from EcoChemie.Autolab.Sdk import Instrument
+    from EcoChemie.Autolab.SDK.Extensions import InstrumentExtensions
+
+except:
+    # TODO: Handle import error
```

Comment:
As long as there is no specific error handling, I would recommend to print a debug message and re-raise the exception with `raise`. There is no need to load the driver if the SDK cannot be imported

Reply:
It should be possible to load the driver without raising an exception, otherwise it will not show up in the list of available drivers in the LCRmeter module. Then, a user would not know why the driver is not shown.

Additionally, I realized that in the future there might be different SDK Versions from 2.1 or they might be saved in a different location. To enable setting the SDK path in the .ini aswell, I moved the load_sdk function insde the driver